proto/workload.proto

```diff
@@ -152,11 +157,12 @@ message Workload {
   // * Requests *to* this workload, via the tunnel_protocol, do not need to be de-tunneled.
   bool native_tunnel = 14;
 
-  // Virtual IPs defines a set of virtual IP addresses the workload can be reached at.
-  // Typically these represent Service ClusterIPs.
-  // The key is a NetworkAddress string of the format network/IP, or just an IP if network
-  // of the service desired matches the network of the workload.
-  map<string, PortList> virtual_ips = 15;
+  // Deprecated. Use `services` to map this workload to service resources.
+  map<string, PortList> virtual_ips = 15 [deprecated = true];
+
+  // The services for which this workload is an endpoint.
+  // The key is the NamespacedHostname string of the format namespace/hostname.
+  map<string, PortList> services = 22;
```

Comment:
With this, does that mean we will fetch two resource before a request can be redirected? Currently we only need fetch once

Reply:
Yes, that's right. But to be fair, it's no different than if we were processing the raw k8s resources.

Reply:
I am not sure as two xds requests are both though network

Reply:
@howardjohn any thoughts here?

Reply:
Not sure what you mean? So will try to answer but may be off

In standard XDS we load all Service and workloads at startup, just like Envoy. Envoy has like 8 resources so not sure why 2 is an issue

In "on demand", we would lookup a single Address. The control plane will return the service+workloads associated with that address. Its technically 1 type (Address), and definitely 1 request.

Reply:
With this change, it will fetch two resource on flying. That means a request latency will be client->ztunnel->ztunnel>server round trip plus fetch worklod time + fetch service time. 

>The control plane will return the service+workloads associated with that address. Its technically 1 type (Address), and definitely 1 request.

I do not understand about this, message proto of one of type `Workload` and `Service`, how could we return both with one request

Reply:
>In standard XDS we load all Service and workloads at startup, just like Envoy

Where does ztunnel do standard xds request for full Address? If we do this, isn't ztunnel store all the services workloads of the mesh, how can it be resource saving

Reply:
> I do not understand about this, message proto of one of type Workload and Service, how could we return both with one request

Ztunnel does not request Workload and Service, it requests `Address` which is a oneof of these types

> Where does ztunnel do standard xds request for full Address? If we do this, isn't ztunnel store all the services workloads of the mesh, how can it be resource saving

Ztunnel on-demand is optional and not enabled by default. It can scale to 100k pod cluster with 50mb RAM due to more efficient protocol/implementation compared to Envoy, _without_ on-demand. On-demand is only needed when we want to hyper-scale beyond that.

With on-demand, service+workload is already fetched on request. This PR doesn't change it really. Note there are some control plane optimizations as well -- if we request a Service we will return all workloads associated with it, we always return node-local workloads, etc.